File: cpp/src/fizz-buzz-whizz/RuleResult.cpp
#include "fizz-buzz-whizz/RuleResult.h"

RuleResult::RuleResult(const std::string& result) : result(result) {
}

bool RuleResult::collect(bool matched, const std::string& str) {
  if (matched) result += str;
  return matched;
}

bool RuleResult::collect(bool matched, const RuleResult& rr) {
  return collect(matched, rr.result);
}

const std::string& RuleResult::toString() const {
  return result;
}

File: java/src/main/java/fizz/buzz/whizz/RuleResult.java
package fizz.buzz.whizz;

public class RuleResult {
  public RuleResult() {
    this("");
  }

  public RuleResult(String str) {
    buff = new StringBuilder(str);
  }

  public boolean collect(boolean matched, String str) {
    if (matched) buff.append(str);
    return matched;
  }

  public boolean collect(boolean matched, RuleResult rr) {
    return collect(matched, rr.buff.toString());
  }

  @Override
  public String toString() {
    return buff.toString();
  }

  private StringBuilder buff;
}
